To: vim_dev@googlegroups.com
Subject: Patch 7.3.287
Fcc: outbox
From: Bram Moolenaar <Bram@moolenaar.net>
Mime-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit
------------

Patch 7.3.287
Problem:    Can't compile with MSVC and tiny options.
Solution:   Move variables and #ifdefs. (Sergey Khorev)
Files:	    src/os_win32.c


*** ../vim-7.3.286/src/os_win32.c	2011-08-10 17:07:56.000000000 +0200
--- src/os_win32.c	2011-08-27 15:07:28.000000000 +0200
***************
*** 3419,3426 ****
  {
      DWORD	availableBytes = 0;
      DWORD	i;
-     int		c;
-     char_u	*p;
      int		ret;
      DWORD	len;
      DWORD	toRead;
--- 3419,3424 ----
***************
*** 3479,3484 ****
--- 3477,3484 ----
  	else if (has_mbyte)
  	{
  	    int		l;
+ 	    int		c;
+ 	    char_u	*p;
  
  	    len += *buffer_off;
  	    buffer[len] = NUL;
***************
*** 3558,3566 ****
      int		noread_cnt = 0;
      garray_T	ga;
      int	    delay = 1;
- # ifdef FEAT_MBYTE
      DWORD	buffer_off = 0;	/* valid bytes in buffer[] */
- # endif
  
      SECURITY_ATTRIBUTES saAttr;
  
--- 3558,3564 ----
***************
*** 3777,3790 ****
  
  	if (WaitForSingleObject(pi.hProcess, delay) != WAIT_TIMEOUT)
  	{
! 	    dump_pipe(options, g_hChildStd_OUT_Rd,
! 			&ga, buffer, &buffer_off);
  	    break;
  	}
  
  	++noread_cnt;
! 	dump_pipe(options, g_hChildStd_OUT_Rd,
! 		    &ga, buffer, &buffer_off);
  
  	/* We start waiting for a very short time and then increase it, so
  	 * that we respond quickly when the process is quick, and don't
--- 3775,3786 ----
  
  	if (WaitForSingleObject(pi.hProcess, delay) != WAIT_TIMEOUT)
  	{
! 	    dump_pipe(options, g_hChildStd_OUT_Rd, &ga, buffer, &buffer_off);
  	    break;
  	}
  
  	++noread_cnt;
! 	dump_pipe(options, g_hChildStd_OUT_Rd, &ga, buffer, &buffer_off);
  
  	/* We start waiting for a very short time and then increase it, so
  	 * that we respond quickly when the process is quick, and don't
*** ../vim-7.3.286/src/version.c	2011-08-26 16:12:55.000000000 +0200
--- src/version.c	2011-08-27 15:08:27.000000000 +0200
***************
*** 711,712 ****
--- 711,714 ----
  {   /* Add new patch number below this line */
+ /**/
+     287,
  /**/

-- 
SUPERIMPOSE "England AD 787".  After a few more seconds we hear hoofbeats in
the distance.  They come slowly closer.  Then out of the mist comes KING
ARTHUR followed by a SERVANT who is banging two half coconuts together.
                 "Monty Python and the Holy Grail" PYTHON (MONTY) PICTURES LTD

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///
